### PR TITLE
Add tbump to simplify package versioning

### LINKED ISSUE #442: Add tbump to simplify package versioning process
Currently, we have to find the lines of code in different files and bump their versions manually. To make this process more automated, we can add [`bump2version`](https://github.com/c4urself/bump2version).

UPDATE: Using `tbump` instead

### PR DESCRIPTION
**Pull request by tomvothecoder**
This PR adds `tbump` to the dev environment. [tbump](https://github.com/dmerejkowsky/tbump) will help simplify the package versioning process for us moving forward. 

Additional changes:    
    - Update docs on releasing
    - Delete unused images from docs
    - Update e3sm-diags version in production env

- Closes #442

### CODE REVIEW

File: conda/e3sm_diags_env.yml [hunk 1]
```diff
@@ -20,5 +20,5 @@ dependencies:
   - netcdf4=1.5.6
   # Additional
   # ==============
-  - e3sm_diags=2.5.0rc4
+  - e3sm_diags=2.5.0
```
Comment (tomvothecoder) [inline conda/e3sm_diags_env.yml:23]: Updating this seems to have been missed in the `v2.5.0` release.

**Comment by tomvothecoder** [comment]
Hey @forsyth2, this PR is ready for review. If things look good here, I can merge the `tbump` PRs for the other tools since they are mostly the same as this PR.

**Review by forsyth2 — COMMENTED** [review]
I also reviewed https://github.com/E3SM-Project/zstash/pull/158 and https://github.com/E3SM-Project/zppy/pull/91 along with this.

I reviewed the code. I also tried the `tbump <version> --no-tag` line on all three repos, and it appears to work. I get `error: failed to push some refs to 'git@github.com:tomvothecoder/e3sm_diags.git'` though -- I'm assuming that's just because I'm working off a branch of your branch though.

File: docs/source/dev_guide/releasing-e3sm-diags.rst [hunk 2]
```diff
@@ -5,191 +5,85 @@ How to Prepare a Release
...
-    ::
-
-        __version__ = 'v1.1.0'
-
-5. Change the ``version`` and ``git_rev`` tag in ``conda/meta.yaml``.
-``version`` is what the version of the software will be on Anaconda and
-``git_rev`` is the tag that we'll setup on GitHub in the next section.
+        # Prepend "v" to <version>
+        # For release candidates, append "rc" to <version>
+        git checkout -b v<version>
```
[the remark this answers is not in the document]
Comment (tomvothecoder) [reply docs/source/dev_guide/releasing-e3sm-diags.rst:24]: Thanks for reviewing all at once. The other docs will be updated with the two commands from here.

**Comment by tomvothecoder** [comment]
> I also reviewed [E3SM-Project/zstash#158](https://github.com/E3SM-Project/zstash/pull/158) and [E3SM-Project/zppy#91](https://github.com/E3SM-Project/zppy/pull/91) along with this.
> 
> I reviewed the code. I also tried the `tbump <version> --no-tag` line on all three repos, and it appears to work. I get `error: failed to push some refs to 'git@github.com:tomvothecoder/e3sm_diags.git'` though -- I'm assuming that's just because I'm working off a branch of your branch though.

I don't think you have write permission on my fork. As long as all of the tbump commands worked then we're good to go.

**Pull request merged**